apollo-livedata-support/src/main/java/com/apollographql/apollo/livedata/LiveDataResponse.kt

```diff
@@ -0,0 +1,74 @@
+package com.apollographql.apollo.livedata
+
+import com.apollographql.apollo.api.Error
+import com.apollographql.apollo.api.Response
+
+/**
+ * The LiveDataResponse class maps [Response] to [LiveDataResponse].
+ * There are Success, Failure, UnsuccessfulResponse, Complete sealed type classes.
+ */
+
+@Suppress("unused")
+sealed class LiveDataResponse<out T> {
+
+    /**
+     * API Success response class.
+     *
+     * [data] is an Optional. (There are responses without data)
+     */
+    class Success<out T>(response: Response<T>) : LiveDataResponse<T>() {
+        val operation = response.operation()
+        val data: T? = response.data()
+        val dependentKeys: Set<String> = response.dependentKeys()
+        val fromCache: Boolean = response.fromCache()
```

Comment:
not sure if these `destructuring` are required. Is there any reason to not have just `val response:Response<T>`?

Reply:
:white_check_mark: Ah, I did not think about `destructuring`. It will be fixed to `var`.